### PR TITLE
Fix some issues in bencmarking

### LINKED ISSUE #231: Performance testing: Convert build system to CMake
We should convert the build system for performance testing and profiling to CMake for better maintainability and extensibility.

We need to:
- [x] Add support for switching between building debug and release versions of compiled binaries.
- [x] Be able to build separate binaries with specific configuration (some with release, some with debug and the same one with both release and debug configuration) by providing a workflow to build a single binary (whose source file is passed as an argument) and passing requested build options as parameters.
- [x] Make sure that all programs and templates in `tests-performance/` use the local implementation of Mata (user-level) rather than the version of Mata installed in the system (system-level). We want to be able to quickly test performance of the current build of Mata, not the stable installed version.
- [x] Convert build system using CMake and integrate it in the main CMake for the whole project to allow better compilation options and IDE integration for performance and profiling tests source files.
- [x] Fix running `make` in `tests-performance/src` to properly include `simlib` and `cudd` libraries.
- [x] Build performance and profiling tests in CI/CD using GitHub Actions.
- [x] Fix errors with some input automata.
- [x] Move template source files into folder `tests-performance/src/templates` so that they do not compile and are easily found.
- [x] Improve the perf testing guide to instruct the user to:
    - [x] copy one of the templates from `tests-performance/src/templates/` folder into the `tests-performance/src` under a new name (which will be the name of the new test binary after compilation) folder (do not forget to specify the importance of the correct target folder in the guide).
    - [x] Prepare an example of copying out one template, renaming the template, modifying the new test, compiling the test, run the test manually and with pycobench.
    - [x] Fix step 4: `as follows:` without any example following.
- [x] Add command `make test` (or similar) which will give some rudimentary results for performance testing.

### PR DESCRIPTION
**Pull request by tfiedor**
This PR:
  1. Fixes issue with not showing any metrics for timeouts;
  2. Adds summary of timeouts to output;
  3. Adds sorting to glob()ed fiels
  4. Fixes some issue with timing macros (wrong comments, no flushing)
  5. Fixes issue with testing coverage testing
  6. Fixes some other minor issues
  7. Write some stuff in readme.

### CODE REVIEW

**Review by Adda0 — APPROVED** [review]
LGTM. Works on Linux as intended.

Could we maybe generate these CSVs into a subfolder which will be ignored in Git?
```
?? tests-integration/corr-double-param-jobs.csv
?? tests-integration/corr-single-param-jobs.csv
```

Furthermore, Codacy has a few good suggestions. They should be applied, too.

File: tests-integration/README.md [hunk 1]
```diff
@@ -64,7 +64,24 @@ In order to test or measure performance of the code you just built by the above
  
 ### How to benchmark and reproduce some of our results
 
-We assume, this is executed from `tests-integration` directory.
+We assume this is executed from `tests-integration` directory.
+
+We recommend to use our `./scripts/run_pyco.sh` scripts with is called as follows:
```
Comment (Adda0) [inline tests-integration/README.md]: Fix typo
```suggestion
We recommend to use our `./scripts/run_pyco.sh` scripts which is called as follows:
```

**Comment by tfiedor** [comment]
@Adda0 I think everything should be addressed now. Have I forgotten anything? If not, you can merge.

**Comment by Adda0** [comment]
All seems well. I am merging the PR.

**Pull request merged**